server/services/socket.js

```diff
@@ -9,12 +9,20 @@ exports.setupSocket = (server, session) => {
     io = socketio(server);
     io.use(sharedsession(session));
     io.on('connection', function (socket) {
-        sockets.push(socket);
-        emitPlayers(gameLogic.getPlayers());
+        if (socket.handshake.session.user) {
+            if (sockets.some(other => other.handshake.session.user === socket.handshake.session.user)) {
```

Comment:
Realised after we did this that you don't need this if statement, just do the filter

Reply:
Do you not? Wouldn't it add an undefined socket to the list?